native/SampleApps/SmartSyncExplorer/RecentContactsTodayExtension/TodayViewController.m

```diff
@@ -0,0 +1,129 @@
+#import <SalesforceSDKCore/NSUserDefaults+SFAdditions.h>
+#import <SalesforceSDKCore/SFRestRequest.h>
+#import <SalesforceSDKCore/SFRestAPI.h>
+#import <SalesforceSDKCore/SFLoggerMacros.h>
+#import <SalesforceSDKCore/SFLogger.h>
+#import <SmartStore/SFQuerySpec.h>
+
+static int const kNumberOfRecords = 3;
+
+@interface TodayViewController () <NCWidgetProviding,UITableViewDelegate,UITableViewDataSource>{
+      NSMutableArray *_contacts;
+}
+@property (weak, nonatomic) IBOutlet UITableView *todayTableView;
+@property (nonatomic, strong) SObjectDataManager *dataMgr;
+
+@end
+
+@implementation TodayViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    #if defined(DEBUG)
+        [SFLogger sharedLogger].logLevel = SFLogLevelDebug;
+    #else
+        [SFLogger sharedLogger].logLevel = SFLogLevelInfo;
+    #endif
+
+    _contacts = [NSMutableArray new];
+    [self.todayTableView setDataSource:self];
+    [self.todayTableView setDelegate:self];
+}
+
+- (void) refreshList {
+    [self.todayTableView reloadData];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+- (void)widgetPerformUpdateWithCompletionHandler:(void (^)(NCUpdateResult))completionHandler {
+    
+    SmartSyncExplorerConfig *config = [SmartSyncExplorerConfig sharedInstance];
+
+    [SFSDKDatasharingHelper sharedInstance].appGroupName = config.appGroupName;
+    [SFSDKDatasharingHelper sharedInstance].appGroupEnabled = YES;
+    if( [self userLoginStatus] ) {
+        [self log:SFLogLevelError format:@"User has logged in"];
+        [SalesforceSDKManager setInstanceClass:[SalesforceSDKManagerWithSmartStore class]];
+        [SalesforceSDKManager sharedManager].connectedAppId = config.remoteAccessConsumerKey;
+        [SalesforceSDKManager sharedManager].connectedAppCallbackUri = config.oauthRedirectURI;
+        [SalesforceSDKManager sharedManager].authScopes = config.oauthScopes;
+        [SalesforceSDKManager sharedManager].authenticateAtLaunch = config.appGroupsEnabled;
+        
+        NSArray *accounts = [[SFUserAccountManager sharedInstance] allUserAccounts];
+        if(accounts && accounts.count>0) {
+            // check for actual Account or id of account or pick the first one?
+            [self log:SFLogLevelError format:@"Loaded user account"];
+            [[SFUserAccountManager sharedInstance] setCurrentUser:accounts[0]];
+            
+            __weak typeof(self) weakSelf = self;
+            void (^completionBlock)(void) = ^{
+                [weakSelf refreshList];
+            };
+          
+            if (!self.dataMgr) {
+                self.dataMgr = [[SObjectDataManager alloc] initWithDataSpec:[ContactSObjectData dataSpec]];
+            }
+            [self.dataMgr lastModifiedRecords:kNumberOfRecords completion:completionBlock];
+            
+        }
+    }
+    completionHandler(NCUpdateResultNewData);
+}
+
+- (BOOL)userLoginStatus {
```

Comment:
This is a little bit of an odd name for its purpose.  Can we simply call this `userIsLoggedIn`?